# **PRAKTIKUM PERTEMUAN 2 DATA SCIENCE**

Nama : Ninis Indah Trisia

NIM : 250401020094

Kelas : IF405

In [ ]:
import pandas as pd
import numpy as np

In [ ]:
url = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"
df = pd.read_csv(url)

In [ ]:
df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


## Pertanyaan Latihan

### Q1 Berapa jumlah baris dan kolom pada dataset Titanic? Kolom apa saja yang tersedia?

In [ ]:
print(df.shape)
print(df.columns)

(891, 12)
Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')


Jadi jumlah baris pada dataset Titanic adalah 891 dan jumlah kolomnya adalah 12.

### Q2 Kolom mana saja yang memiliki missing values? Berapa jumlah dan proporsinya (%) per kolom?

In [ ]:
missing = df.isnull().sum()
persentase = (missing / len(df)) * 100

hasil_missing = pd.DataFrame({
    'Jumlah Missing': missing,
    'Persentase (%)': persentase
})

print(hasil_missing)

             Jumlah Missing  Persentase (%)
PassengerId               0        0.000000
Survived                  0        0.000000
Pclass                    0        0.000000
Name                      0        0.000000
Sex                       0        0.000000
Age                     177       19.865320
SibSp                     0        0.000000
Parch                     0        0.000000
Ticket                    0        0.000000
Fare                      0        0.000000
Cabin                   687       77.104377
Embarked                  2        0.224467


Kolom yang memiliki missing value adalah sebagai berikut:

*   Kolom age dengan jumlah missing 177 dan
persentase 19.865320%
*   Kolom cabin dengan jumlah missing 687 dan
persentase 77.104377%
*   Kolom embarked dengan jumlah missing 2 dan
persentase 0.224467%





### Q3 Berapa persen penumpang yang selamat (Survived=1)? Gunakan operasi NumPy atau Pandas.

In [ ]:
persen_selamat = df['Survived'].mean() * 100
print(f"Persentase penumpang selamat: {persen_selamat:.2f}%")

Persentase penumpang selamat: 38.38%


Jadi persentase penumpang selamat ada 38.38%

### Q4 Filter penumpang wanita dari kelas 1 (Pclass=1). Berapa jumlahnya dan berapa rata-rata usianya?

In [ ]:
wanita_kelas1 = df[(df['Sex'] == 'female') & (df['Pclass'] == 1)]

jumlah = len(wanita_kelas1)
rata_usia = wanita_kelas1['Age'].mean()

print("Jumlah:", jumlah)
print("Rata-rata usia:", rata_usia)

Jumlah: 94
Rata-rata usia: 34.61176470588235


Berdasarkan filter penumpang wanita dari kelas 1 (Pclass=1), jumlah penumpang wanita adalah 94 dengan rata-rata usia 34.61176470588235

### Q5 Gunakan groupby untuk menghitung tingkat keselamatan (mean Survived) per kelas penumpang. Kelas mana yang paling tinggi?

In [ ]:
keselamatan = df.groupby('Pclass')['Survived'].mean()
print(keselamatan)

Pclass
1    0.629630
2    0.472826
3    0.242363
Name: Survived, dtype: float64


Jadi hasil groupby, tingkat keselamatan tertinggi ada pada (Pclass) 1 dengan tingkat keselamatan 0.629630

### Q6 Buat analisis tambahan berdasarkan variabel lain yang Anda anggap menarik. Sertakan interpretasi singkat.

In [ ]:
fare_kelas = df.groupby('Pclass')['Fare'].mean()
print(fare_kelas)

Pclass
1    84.154687
2    20.662183
3    13.675550
Name: Fare, dtype: float64


Analisis tambahan yang saya sertakan adalah menghitung rata-rata tarif tike (Fare) berdasarkan kelas penumpang (Pclass). Hasil perhitungan menunjukkan bahwa kelas 1 memiliki rata-rata tarif tertinggi yaitu 84.154687%

##Kesimpulan

Pada praktikum ini saya mempelajari dasar-dasar pengolahan dan analisis data menggunakan Python dengan memanfaatkan library Pandas dan NumPy. Dataset yang digunakan adalah dataset Titanic yang berisi informasi mengenai penumpang, seperti kelas penumpang, usia, jenis kelamin, tarif tiket, dan status keselamatan.

Dari hasil eksplorasi data diketahui bahwa dataset terdiri dari 891 baris dan 12 kolom. Selain itu, ditemukan beberapa kolom yang memiliki missing values, terutama pada kolom Cabin dan Age, sehingga menunjukkan bahwa kualitas data perlu diperhatikan sebelum digunakan untuk analisis yang lebih lanjut.

Analisis yang dilakukan menunjukkan bahwa sekitar 38,38% penumpang berhasil selamat dari tragedi Titanic. Ketika data difilter berdasarkan jenis kelamin dan kelas penumpang, ditemukan bahwa terdapat 94 penumpang wanita di kelas 1 dengan rata-rata usia sekitar 34,6 tahun. Hasil ini memberikan gambaran mengenai karakteristik penumpang pada kelompok tertentu.

Melalui penggunaan fungsi groupby, diketahui bahwa penumpang kelas 1 memiliki tingkat keselamatan tertinggi dibandingkan kelas lainnya. Selain itu, analisis tambahan terhadap tarif tiket menunjukkan bahwa rata-rata harga tiket tertinggi juga dimiliki oleh penumpang kelas 1. Hal ini mengindikasikan adanya hubungan antara kelas penumpang dengan fasilitas yang diperoleh maupun peluang keselamatan selama kejadian.

Secara keseluruhan, praktikum ini membantu saya memahami cara membaca dataset, mengidentifikasi data yang hilang, melakukan penyaringan data, menghitung statistik sederhana, serta menarik informasi dari data menggunakan fungsi-fungsi dasar Pandas. Dari kegiatan ini saya juga belajar bahwa analisis data dapat digunakan untuk menemukan pola dan informasi yang tidak langsung terlihat hanya dari melihat data mentah.